src/app/CommandSender.h

```diff
@@ -50,9 +50,12 @@ class CommandSender : public Command, public Messaging::ExchangeDelegate
     // handle calling Shutdown on itself once it decides it's done with waiting
     // for a response (i.e. times out or gets a response).
     //
+    // Client can specify the maximum time to wait for response (in milliseconds) via timeout parameter.
+    // Default timeout value will be used otherwise.
+    //
     // If SendCommandRequest is never called, or the call fails, the API
     // consumer is responsible for calling Shutdown on the CommandSender.
-    CHIP_ERROR SendCommandRequest(NodeId aNodeId, FabricIndex aFabricIndex, SessionHandle * secureSession = nullptr);
+    CHIP_ERROR SendCommandRequest(NodeId aNodeId, FabricIndex aFabricIndex, SessionHandle * secureSession, uint32_t timeout = 0);
```

Comment:
Instead of defining 0 is a magic value that results in using the default `kImMessageTimeoutMsec`', why not instead just set `kImMessageTimeoutMsec` as the default here?

Reply:
Acked, the reason I do this trick is there are some conflicts if I include #include <app/InteractionModelEngine.h> directly in those header files use kImMessageTimeoutMsec.

This is resolved by declaring kImMessageTimeoutMsec in InteractionModelDelegate.h which already is the common header files